open-build-service-mock/src/lib.rs

```diff
@@ -704,6 +704,9 @@ impl ObsMock {
         let mut projects = self.inner.projects.write().unwrap();
         let project = get_project(&mut *projects, project_name);
 
+        // Make sure the source package exists.
+        get_package(project, &package_name);
```

Comment:
And the effect of this is that it panics if the package is not there? Wouldn't that be more natural to do later on when working out the package (e.g. if there is no entry for the package_name)?

Reply:
If I'm understanding correctly, the reason I added this here is the split storage thing, where package source info is stored directly in a map under the project, but build-related information is stored in the separate repositories map (to accommodate OBS sometimes deciding package build info is "inside" the source package and sometimes decided it's "inside" the repo). However, the build package side afaik simply cannot exist without the source package, but the source package side *can* exist without the build package side?

i.e. having package build info with no corresponding source package will break other parts of the mock crate, but the other way around is entirely fine. So, I need to make sure there's always a source package, but the build package info can safely be inserted on-demand.

Reply:
In that case why not assert project.packages.contains_key(name) here then (potentially in a helper); It's quite confusing to see a getter being used with the result not used, apparently only for the side-effect that it will panic on failure (hence the first part of the question)

Fwiw i'm not entirely convinced creating the build result on demand by this function makes sense (shouldn't it only exist after a build has done); but i can go with that part for now

Reply:
I was just being way too allergic to insanely marginal copy-pasting; changed it to an assert now.

> Fwiw i'm not entirely convinced creating the build result on demand by this function makes sense (shouldn't it only exist after a build has done); but i can go with that part for now

Honestly I'm thinking I made the wrong call when I split the "build" and "source" sides early on, since they're a lot more tied together than I had expected, and it's certainly resulted in its own set of hacks to deal with. (That being said, I may just not be fully familiarized with the hacks that would be required to unify these.)

Reply:
ah i missed the outdated note :) not that used to github